src/include/index/vamana_index.h

```diff
@@ -179,14 +180,16 @@ class vamana_index {
       size_t num_nodes,
       size_t l_build,
       size_t r_max_degree,
-      std::optional<TemporalPolicy> temporal_policy = std::nullopt)
+      std::optional<TemporalPolicy> temporal_policy = std::nullopt,
+      DistanceMetric distance_metric = DistanceMetric::L2)
```

Comment:
Can we pass the distance function here and instantiate all the different values in dispatch_tables?